cap-primitives/src/std/fs/open_manually.rs

```diff
@@ -141,18 +141,14 @@ pub(crate) fn open_manually(
                         dirs.push(prev_base);
                         canonical_path.push(one);
                     }
-                    Err(e) => match e.raw_os_error() {
-                        Some(libc::ELOOP) | Some(libc::EMLINK) if use_options.nofollow => {
-                            return Err(io::Error::from_raw_os_error(libc::ELOOP))
-                        }
-                        Some(libc::ELOOP) | Some(libc::EMLINK) => {
-                            let destination =
-                                resolve_symlink_at(base.as_file(), &one, symlink_count)?;
-                            components
-                                .extend(destination.components().map(to_owned_component).rev());
-                        }
-                        _ => return Err(e),
-                    },
+                    Err(OpenUncheckedError::Symlink) if use_options.nofollow => {
+                        return symlink_disallowed()
```

Comment:
Having the original `io::Error` would be useful here too, so that we can forward on the OS's original error message.